### PR TITLE
Basic rss interface

### PR DESCRIPTION
**Pull request by will-blk**
# Rss basic interface
- Enable/Disable background processing option
- Enable/Disable autodownload articles option
- Create/Delete feeds
- Create/Delete rules (Only supports "must contain" input and affected feed)

# PR Checklist
- [x] I've started from master
- [x] I've only committed changes related to this PR
- [x] All Unit tests pass
- [x] I've removed all commented code
- [x] I've removed all unneeded console.log statements

Issue #311 resolves

### CODE REVIEW

File: src/lang/en.js [hunk 1]
```diff
@@ -191,6 +203,29 @@ const locale = {
         whenRatioReaches: 'When ratio reaches',
         whenSeedingTimeReaches: 'When seeding time reaches'
       },
+      pageRss: {
+        tabName: {
+          general: 'Geral',
```
Comment (WDaan) [inline src/lang/en.js:208]: this seems like a typo :p

Comment (WDaan) [reply src/lang/en.js:208]: Also, I think you need to add 'feed' at the top here in the 'General' section... at the top of the file

File: src/services/qbit.js [hunk 2]
```diff
@@ -154,6 +154,42 @@ class Qbit {
...
+
+  createRule(ruleName, defs) {
+    return this.execute('post', '/rss/setRule', {
+      ruleName: ruleName,
+      ruleDef: JSON.stringify(defs)
+    })
+  }
+
+  getFeeds() {
+    return this.axios.get('/rss/items').then(res => res.data)
```
Comment (WDaan) [inline src/services/qbit.js:174]: I'd sanitize the data here 

```
    return this.axios.get('/rss/items')
      .then(res => res.data)
      .then(data => 
        Object.entries(data).map(feed => {
          return { name: feed[0], ...feed[1] }
        })
      )
```


Same for rules 
```
  getRules() {
    return this.axios.get('/rss/rules')
      .then(res => res.data)
      .then(data => Object.entries(data).map(rule => {
        return { name: rule[0], ...rule[1] }
      }))
  }
```

File: src/store/mutations.js [hunk 3]
```diff
@@ -80,6 +80,18 @@ export default {
     state.sort_options.tracker = tracker
   },
   FETCH_CATEGORIES: async state => state.categories = Object.values(await (qbit.getCategories())),
+  FETCH_FEEDS: async state => {
+    const feeds = await qbit.getFeeds()
+    state.feeds = Object.entries(feeds).map(feed => { 
+      return { name: feed[0], ...feed[1] }
+    })
+  },
+  FETCH_RULES: async state => {
```
Comment (WDaan) [inline src/store/mutations.js:89]: we can keep this clean if we sanitize/format the data when doing the api call

```
  FETCH_FEEDS: async state => state.rss.feeds = await qbit.getFeeds(),
  FETCH_RULES: async state => state.rss.rules = await qbit.getRules(),
```

**Comment by WDaan** [comment]
Awesome work! Seems to be working perfectly :) 

I've commented some minor changes, I can fix them myself if you want.

Create/Delete on Rules is okay... but we should support the full form in the future
Also the feed multi-select part in the create rule will not be ideal for people with a lot of feeds, but I guess that's also a later improvement.


Great work so far :) thanks

File: src/lang/en.js (hunk 1 above)
Comment (will-blk) [reply src/lang/en.js:208]: A mix of english and portuguese hahaha Will fix it

**Comment by will-blk** [comment]
Thanks :smile: I will make the fixes soon, haven't seen the store index file before. 

As an improvement idea, I would also move RSS from the settings and make a tab elsewhere. I think It will become too small after adding the feed preview...

**Comment by will-blk** [comment]
Check it out!

**Pull request merged**